InOut/CMakeLists.txt

```diff
@@ -169,24 +170,22 @@ else()
 endif()
 
 if(USE_PORTMIDI)
-    set(pmidi_LIBS ${PORTMIDI_LIBRARY})
     include_directories(${PORTMIDI_INCLUDE_PATH})
+	make_plugin(pmidi pmidi.c)
     if(LINUX)
         if(NEED_PORTTIME)
-           list(APPEND pmidi_LIBS ${PORTTIME_LIBRARY})
-           include_directories(${PORTTIME_INCLUDE_PATH})
-        endif()
-    endif()
-    if(WIN32)
-      if(NEED_PORTTIME)
-        include_directories(${PORTTIME_INCLUDE_PATH})
-      endif()
-        list(APPEND pmidi_LIBS ${CSOUND_WINDOWS_LIBRARIES})
-    endif()
-    if(LINUX AND ALSA_LIBRARY)
-        list(APPEND pmidi_LIBS ${ALSA_LIBRARY})
+			include_directories(${PORTTIME_INCLUDE_PATH})
+			target_link_libraries(pmidi ${PORTTIME_LIBRARY})
+		endif()
+    elseif(WIN32)
+		if(NEED_PORTTIME)
+			include_directories(${PORTTIME_INCLUDE_PATH})
+		endif()
+		target_link_libraries(pmidi ${CSOUND_WINDOWS_LIBRARIES})
+    elseif(LINUX AND ALSA_LIBRARY)
+		target_link_libraries(pmidi ${ALSA_LIBRARY})
     endif()
-    make_plugin(pmidi pmidi.c "${pmidi_LIBS}")
+    target_link_libraries(pmidi ${PORTMIDI_LIBRARY})

```

Comment:
I think we may have talked about this on Slack, but we should review our use of target_link_libraries and make sure it's consistent: if we're going to do it in some places, might as well do it everywhere.